src/navier_stokes/INSStaggeredHierarchyIntegrator.cpp

```diff
@@ -2287,27 +2384,43 @@ INSStaggeredHierarchyIntegrator::reinitializeOperatorsAndSolvers(const double cu
     const double rho = d_problem_coefs.getRho();
     const double mu = d_problem_coefs.getMu();
     const double lambda = d_problem_coefs.getLambda();
-    double K = 0.0;
+    double K1 = 1.0, K2 = 0.0;
+    const double omega = getTimeStepSizeRatio();
     switch (d_viscous_time_stepping_type)
     {
     case BACKWARD_EULER:
-        K = 1.0;
+        K1 = 1.0;
+        K2 = 1.0;
+        break;
+    case BDF2:
+        if (getIntegratorStep() == 0)
+        {
+            K1 = 1.0;
+            K2 = 1.0;
+        }
+        else
+        {
+            K1 = (1.0 + 2.0 * omega) / (1.0 + omega);
+            K2 = 1.0;
+        }
         break;
     case FORWARD_EULER:
-        K = 0.0;
+        K1 = 1.0;
+        K2 = 0.0;
         break;
     case TRAPEZOIDAL_RULE:
-        K = 0.5;
+        K1 = 1.0;
+        K2 = 0.5;
         break;
     default:
         TBOX_ERROR("this statment should not be reached");
     }
     PoissonSpecifications U_problem_coefs(d_object_name + "::U_problem_coefs");
-    U_problem_coefs.setCConstant((rho / dt) + K * lambda);
-    U_problem_coefs.setDConstant(-K * mu);
+    U_problem_coefs.setCConstant(K1 * rho / dt + K2 * lambda);
+    U_problem_coefs.setDConstant(-K2 * mu);
     PoissonSpecifications P_problem_coefs(d_object_name + "::P_problem_coefs");
     P_problem_coefs.setCZero();
-    P_problem_coefs.setDConstant(rho == 0.0 ? -1.0 : -1.0 / rho);
+    P_problem_coefs.setDConstant(rho == 0.0 ? -1.0 : -1.0 / (K1 * rho));
```

Comment:
P is not timestepped.
```suggestion
    P_problem_coefs.setDConstant(rho == 0.0 ? -1.0 : -1.0 / rho);
```

Reply:
Here I have to admit that I have forgotten why we need to include `rho` in the pressure setup.

Reply:
Oh hmmm... I think I misunderstood what this was being used for.

Reply:
Specifically, at least for the test I am running, this seems to make no difference for `TRAPEZOIDAL_RULE` but to shave one iteration off for `BDF2`. Let me see if I can figure out whether we care about this or not in practice.

Reply:
This is a little obscure, but the projection preconditioner is derived from
```
    //    rho (U - U^*) + G Phi = 0
    //    -D U = F_P
```
and then figuring out how `Phi` relates to `P`. I think that for a BDF scheme, we need to do instead
```
    //    K1 rho (U - U^*) + G Phi = 0
    //    -D U = F_P
```
where `K1` is this factor that is hitting `U^{n+1}` in the BDF time stepping scheme.